### ISSUE TITLE
All groups and tags suddenly lost

### ISSUE THREAD

**Issue by swekia** [JabRef/jabref#7442]
JabRef 5.2--2020-12-24--6a2a512
Windows 10 10.0 amd64 
Java 14.0.2

Hi,

I just discovered that all the groups that I've created (many, many hours of manual work) disappeared. I haven't been using JabRef much after I updated to 5.2, so I cannot say for sure whether this issue must be related to updating.

About the groups/tags:
I had many groups, based on keywords, explicit selection and free search expressions. Now there's no group left. In the bibtex file, there are some `groups   = {Keywords}` if any (which is not useful at all).

To be honest, I'm really angry at the moment as I'm not using JabRef for fun but really need it for work and I cannot waste that much time on recreating tagging, grouping etc. over and over again. Some months ago, I had a similar problem with the "read status" that got lost arbitrarily. At the moment, I've no trust in JabRef's reliability anymore.

What can I do now?

Thanks in advance and best regards

**Comment by koppor** [comment]
@swekia Do you have an old version of the file? Maybe a `.bak` file? What is at the end of the `.bib` file? It should look as follows: https://github.com/JabRef/jabref/blob/f26e25b7bedbbae81f3bec8b30dd482aba3d251c/src/test/resources/testbib/jabref-authors.bib#L3008

**Comment by swekia** [comment]
Yes, I'm lucky that I could restore my old bib file from a backup.
Nonetheless, I think it shouldn't be necessary to do that; especially, as not everyone is backing up.

**Comment by calixtus** [comment]
Thanks for your issue report. This sounds indeed like a serious problem.

Can you provide us with some more information, when this is happening, what action leads to loosing you groups? When did you notice?

Would it be possible for you to take a quick look into your bib file and tell us, if there is still a `groups = {...}` line included in your bibentries and at the end of the file if there is something like this:
```
@Comment{jabref-meta: grouping:
0 AllEntriesGroup:;
1 StaticGroup:SomeGroupName\;0\;1\;0x8a8a8aff\;\;\;;
```
Do you have any other programs running working on the same library at the same time, like a texeditor?
Did JabRef show a warning about the library having changed?

**Comment by swekia** [comment]
Thank you for your reply. Actually, I cannot say if this is happening on a regular basis. I noticed it twice but for the first time this happened, I cannot say when exactly that was. For the second this it was just the day after I restored my file from backup.

Yes, currently, the groups are fine, this is an excerpt of the bibtex file:

```
@Article{riedhammer2012revisiting,
  author     = {Riedhammer, Korbinian and Bocklet, Tobias and Ghoshal, Arnab and Povey, Daniel},
  journal    = {2012 IEEE International Conference on Acoustics, Speech and Signal Processing (ICASSP)},
  title      = {Revisiting semi-continuous hidden Markov models},
  year       = {2012},
  pages      = {4721--4724},
  file       = {:kaldi/riedhammer2012revisiting.pdf:PDF},
  groups     = {Keywords},
  keywords   = {automatic speech recognition; acoustic model; kaldi; hidden markov models},
  readstatus = {skimmed},
}

...

@Comment{jabref-meta: grouping:
0 AllEntriesGroup:;
1 StaticGroup:Keywords\;0\;1\;0x8a8a8aff\;\;group from keywords\;;
...
2 KeywordGroup:Automatic Speech Recognition\;2\;keywords\;automatic speech recognition\;0\;0\;1\;0x8a8a8aff\;\;\;;
3 KeywordGroup:Speech Recognition\;0\;keywords\;speech recognition\;0\;0\;1\;0x8a8a8aff\;\;\;;
3 KeywordGroup:Word Recognition\;0\;keywords\;word recognition\;0\;0\;1\;0x8a8a8aff\;\;\;;
...
2 KeywordGroup:partly\;0\;readstatus\;partly\;0\;0\;1\;0x8a8a8aff\;\;\;;
2 KeywordGroup:read\;0\;readstatus\;read\;0\;0\;1\;0x8a8a8aff\;\;set readstatus to "read"\;;
1 SearchGroup:Books\;0\;entrytype=book\;0\;0\;1\;\;\;\;;
}
```
(I put the `...` for shortening.) 

I use old-fashioned TeXnicCenter but only for compiling the bibliography, so it *shouldn't* change anything in there. I've checked that the "last modified" date of the `.bib` file doesn't change when compiling with TeXnicCenter. 

Is it possible that this problem is related to exporting the bib-file? I found that "Save selected as plain BibTeX" seems to definitely remove all JabRef metadata. Is that on purpose?

Usually, I export only the files I need for my current project to a separate `.bib` file, so I don't overwrite the big one. So to my current understanding, everytime I change something in JabRef and press `Ctrl + S`, the `.bib` file is updated with the latest changes, isn't it? And when I export selected entries as plain BibTeX, I export the single items only. Is that correct?

**Comment by calixtus** [comment]
I thought we deactivated the stale bot... Sorry.

**Comment by geyslein** [comment]
HI guys, the issue is quite old but still open, so.....I ran into the same problem today. Don't know if you sorted out the reason why this happens. I have no clue either. Just one thing. In my bib file everything meta after `@Comment{jabref-meta: databaseType:bibtex;}` was deleted. I didn't export/import/open/closed bib files as you mentioned here. I didn't do something fancy with my database either.

**Comment by Siedlerchr** [comment]
@geyslein  Please check your backups https://docs.jabref.org/advanced/autosave#where-can-i-find-the-backup-files

**Comment by geyslein** [comment]
Thanks @Siedlerchr I had no probs earlier to recover the groups through the backups. Just  wanted to point out, that the issue still exists

**Comment by koppor** [comment]
> HI guys, the issue is quite old but still open, so.....I ran into the same problem today. Don't know if you sorted out the reason why this happens. I have no clue either. Just one thing. In my bib file everything meta after `@Comment{jabref-meta: databaseType:bibtex;}` was deleted. I didn't export/import/open/closed bib files as you mentioned here. I didn't do something fancy with my database either.

By some chance, do you have logs of that day available?

Here on Windows:

    C:\Users\koppor\AppData\Local\org.jabref\jabref\logs\6.0--2024-12-04--0dad3e1

**Comment by geyslein** [comment]
hey @koppor 
I attached you the logs of the time span I suspect. Never happened again since....

[log_2025-02-16_17-28-04.txt](https://github.com/user-attachments/files/19360887/log_2025-02-16_17-28-04.txt)
[log_2025-02-16_17-24-38.txt](https://github.com/user-attachments/files/19360886/log_2025-02-16_17-24-38.txt)
[log_2025-02-15_16-22-19.txt](https://github.com/user-attachments/files/19360888/log_2025-02-15_16-22-19.txt)
[log_2025-02-10_14-27-24.txt](https://github.com/user-attachments/files/19360885/log_2025-02-10_14-27-24.txt)
